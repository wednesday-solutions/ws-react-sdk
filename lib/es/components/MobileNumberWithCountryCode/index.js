import{_ as e,b as t,d as r}from"../_rollupPluginBabelHelpers-c2a46d82.js";import"../_rollup-plugin-inject-process-env-1259e43d.js";import n,{useState as o,useEffect as i}from"react";import a from"prop-types";import{u as m}from"../cognito-cd75d0f5.js";import l from"libphonenumber-js";import{FormattedMessage as s}from"react-intl";import d from"../CustomInput/index.js";import u from"../FormItem/index.js";import{appIntl as p}from"../IntlGlobalProvider/index.js";import f from"../If/index.js";import c from"../T/index.js";import g from"styled-components";import{c as b}from"../colors-93a064d8.js";import{f as F}from"../fonts-ca6864a0.js";import{m as y}from"../media-ee9cbc51.js";import"amazon-cognito-identity-js";import"antd/lib/input/style";import"antd/lib/input";import"antd/lib/form/style";import"antd/lib/form";var h,v,C,_,x,V=e=>((e,t,r)=>({[e]:t,message:r.trim().length>0?n.createElement(s,{id:r}):"Invalid Input Value"}))("required",!0,e),j=(e,t)=>{try{return(null==t||!t.length||"0"!==t[0])&&l(e+t).isValid()}catch(e){return!1}},w=e=>{var t=e;return e&&"+"!==e[0]&&(t="+"+e),t},E=(e,t)=>{var r=t;return t&&"0"===t[0]&&(r=t.substr(1,t.length)),w(e)+r},k=g(u)(h||(h=e(["\n  flex: 0.25;\n  ",";\n  max-width: 6rem;\n  min-width: 4rem;\n"])),y.tablet.max("\n  flex: 0.15;\n  ")),I=g.div(v||(v=e(["\n  display: flex;\n"]))),M=g(u)(C||(C=e(["\n  flex: 1;\n  margin-left: 1rem !important;\n"]))),T=g(c)(_||(_=e(["\n  && {\n    cursor: pointer;\n    color: ",";\n    text-transform: uppercase;\n    ",";\n    ",";\n    margin-bottom: 0;\n  }\n"])),b.darkFuchsia,F.size.xSmall(),F.weights.fw600()),L=g(d)(x||(x=e(["\n  /* Chrome, Safari, Edge, Opera */\n  input::-webkit-outer-spin-button,\n  input::-webkit-inner-spin-button {\n    -webkit-appearance: none;\n    margin: 0;\n  }\n  /* Firefox */\n  input[type='number'] {\n    -moz-appearance: textfield;\n  }\n"])));function N(e){var{countryCode:a,strippedNumber:l,showVerify:s}=e,u=t(e,["countryCode","strippedNumber","showVerify"]),[c,g]=o(!1),[b,F]=o(!1),[y,h]=o();i((()=>{y||E(a,l)===y||F(!1)}),[l,a]);var v,C=()=>{c?E(u.getForm().getFieldValue("countryCode"),u.getForm().getFieldValue("phone"))===y&&g(!1):(g(!0),h(E(a,l)))},_=function(){var e=r((function*(){if(j(u.getForm().getFieldValue("countryCode"),u.getForm().getFieldValue("phone")))yield m(u.getForm().getFieldValue("email"),E(u.getForm().getFieldValue("countryCode"),u.getForm().getFieldValue("phone"))),u.verifyClickListener();else{u.getForm().validateFields(["countryCode","phone"])}}));return function(){return e.apply(this,arguments)}}();return n.createElement(I,{"data-testid":"mobile-number-with-country-code"},n.createElement(k,{initialValue:w(a),rules:[{message:p().formatMessage({id:"invalid_mobile_num_msg"}),validator:(e,t,r)=>{j(u.getForm().getFieldValue("countryCode"),u.getForm().getFieldValue("phone"))?r():u.getForm().setFields([{name:"phone",errors:[p().formatMessage({id:"invalid_mobile_num_msg"})]}])}}],validateTrigger:"onBlur",name:"countryCode"},n.createElement(d,{onChange:C,"data-testid":"code-input",autoComplete:"off",maxLength:4,placeholder:process.env.DEFAULT_COUNTRY_CODE})),n.createElement(M,{initialValue:l,rules:[...(v=p().formatMessage({id:"field_required"}),[V(v)]),{message:p().formatMessage({id:"invalid_mobile_num_msg"}),validator:(e,t,r)=>{j(u.getForm().getFieldValue("countryCode"),u.getForm().getFieldValue("phone"))?r():r(new Error(p().formatMessage({id:"invalid_mobile_num_msg"})))}}],validateTrigger:"onBlur",name:"phone"},n.createElement(L,{type:"number",suffix:n.createElement(f,{condition:!b&&s&&c},n.createElement(T,{id:"verify",onClick:_})),onChange:C,autoComplete:"off","data-testid":"mobile-input",placeholder:p().formatMessage({id:"contact_no"})})))}N.propTypes={getForm:a.func,countryCode:a.string,strippedNumber:a.string,showVerify:a.bool,verifyClickListener:a.func},N.defaultProps={verifyClickListener:()=>{}};export default N;
