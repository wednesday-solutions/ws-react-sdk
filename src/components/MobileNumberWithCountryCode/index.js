/**
 *
 * MobileNumberWithCountryCode
 *
 */

import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { updatePhoneNumberRequest } from '@services/cognito';
import { isRequired, isValidNumberWithCountry } from '@utils/validationUtils';
import CustomInput from '@components/CustomInput';
import FormItem from '@components/FormItem';
import { appIntl } from '@components/IntlGlobalProvider';
import If from '@components/If';
import T from '@components/T';
import styled from 'styled-components';
import { fonts, media, colors } from '@themes';
import { getValidCountryCode, getValidPhoneNumber } from '@utils/common';

const CountryCode = styled(FormItem)`
  flex: 0.25;
  ${media.tablet.max(`
  flex: 0.15;
  `)};
  max-width: 6rem;
  min-width: 4rem;
`;
const MobileNoContainer = styled.div`
  display: flex;
`;
const MobileNoField = styled(FormItem)`
  flex: 1;
  margin-left: 1rem !important;
`;

const Verify = styled(T)`
  && {
    cursor: pointer;
    color: ${colors.darkFuchsia};
    text-transform: uppercase;
    ${fonts.size.xSmall()};
    ${fonts.weights.fw600()};
    margin-bottom: 0;
  }
`;
const CustomMobileInput = styled(CustomInput)`
  /* Chrome, Safari, Edge, Opera */
  input::-webkit-outer-spin-button,
  input::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }
  /* Firefox */
  input[type='number'] {
    -moz-appearance: textfield;
  }
`;

function MobileNumberWithCountryCode({ countryCode, strippedNumber, showVerify, ...props }) {
  const [numberChanged, setNumberChanged] = useState(false);
  const [requestedVerification, setRequestedVerification] = useState(false);
  const [originalNumber, setOriginalNumber] = useState();
  useEffect(() => {
    if (!originalNumber && getValidPhoneNumber(countryCode, strippedNumber) !== originalNumber) {
      setRequestedVerification(false);
    }
  }, [strippedNumber, countryCode]);
  const handleOnChange = () => {
    if (!numberChanged) {
      setNumberChanged(true);
      setOriginalNumber(getValidPhoneNumber(countryCode, strippedNumber));
    } else if (
      getValidPhoneNumber(props.getForm().getFieldValue('countryCode'), props.getForm().getFieldValue('phone')) ===
      originalNumber
    ) {
      setNumberChanged(false);
    }
  };

  const handleVerifyClickListener = async () => {
    if (
      !isValidNumberWithCountry(props.getForm().getFieldValue('countryCode'), props.getForm().getFieldValue('phone'))
    ) {
      const fields = ['countryCode', 'phone'];
      props.getForm().validateFields(fields);
    } else {
      await updatePhoneNumberRequest(
        props.getForm().getFieldValue('email'),
        getValidPhoneNumber(props.getForm().getFieldValue('countryCode'), props.getForm().getFieldValue('phone'))
      );
      props.verifyClickListener();
    }
  };
  return (
    <MobileNoContainer data-testid={'mobile-number-with-country-code'}>
      <CountryCode
        initialValue={getValidCountryCode(countryCode)}
        rules={[
          {
            message: appIntl().formatMessage({ id: 'invalid_mobile_num_msg' }),
            validator: (rule, value, callback) => {
              if (
                !isValidNumberWithCountry(
                  props.getForm().getFieldValue('countryCode'),
                  props.getForm().getFieldValue('phone')
                )
              ) {
                props.getForm().setFields([
                  {
                    name: 'phone',
                    errors: [
                      appIntl().formatMessage({
                        id: 'invalid_mobile_num_msg'
                      })
                    ]
                  }
                ]);
              } else {
                callback();
              }
            }
          }
        ]}
        validateTrigger={'onBlur'}
        name="countryCode"
      >
        <CustomInput
          onChange={handleOnChange}
          data-testid="code-input"
          autoComplete="off"
          maxLength={4}
          placeholder={process.env.DEFAULT_COUNTRY_CODE}
        />
      </CountryCode>
      <MobileNoField
        initialValue={strippedNumber}
        rules={[
          ...isRequired(appIntl().formatMessage({ id: 'field_required' })),
          {
            message: appIntl().formatMessage({ id: 'invalid_mobile_num_msg' }),
            validator: (rule, value, callback) => {
              if (
                !isValidNumberWithCountry(
                  props.getForm().getFieldValue('countryCode'),
                  props.getForm().getFieldValue('phone')
                )
              ) {
                callback(new Error(appIntl().formatMessage({ id: 'invalid_mobile_num_msg' })));
              } else {
                callback();
              }
            }
          }
        ]}
        validateTrigger={'onBlur'}
        name="phone"
      >
        <CustomMobileInput
          type="number"
          suffix={
            <If condition={!requestedVerification && showVerify && numberChanged}>
              <Verify id="verify" onClick={handleVerifyClickListener} />
            </If>
          }
          onChange={handleOnChange}
          autoComplete="off"
          data-testid="mobile-input"
          placeholder={appIntl().formatMessage({ id: 'contact_no' })}
        />
      </MobileNoField>
    </MobileNoContainer>
  );
}

MobileNumberWithCountryCode.propTypes = {
  getForm: PropTypes.func,
  countryCode: PropTypes.string,
  strippedNumber: PropTypes.string,
  showVerify: PropTypes.bool,
  verifyClickListener: PropTypes.func
};

MobileNumberWithCountryCode.defaultProps = {
  verifyClickListener: () => {}
};

export default MobileNumberWithCountryCode;
