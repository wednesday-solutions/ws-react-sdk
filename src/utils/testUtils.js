import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { DEFAULT_LOCALE, translationMessages } from '@src/i18n';
import IntlGlobalProvider from '@components/IntlGlobalProvider';

export const renderWithIntl = children => {
  return render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );
};

export const getComponentStyles = (Component, props = {}) => {
  renderWithIntl(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);
  // eslint-disable-next-line no-underscore-dangle
  return window.getComputedStyle(componentRoots[0])._values;
};

export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export default {
  timeout,
  renderWithIntl,
  getComponentStyles,
  apiResponseGenerator
};
