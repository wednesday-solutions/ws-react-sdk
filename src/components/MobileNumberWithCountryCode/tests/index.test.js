/**
 *
 * Tests for MobileNumberWithCountryCode
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import MobileNumberWithCountryCode from '../index';

describe('<MobileNumberWithCountryCode />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<MobileNumberWithCountryCode />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 MobileNumberWithCountryCode component', () => {
    const { getAllByTestId } = renderWithIntl(<MobileNumberWithCountryCode />);
    expect(getAllByTestId('mobile-number-with-country-code').length).toBe(1);
  });
});
