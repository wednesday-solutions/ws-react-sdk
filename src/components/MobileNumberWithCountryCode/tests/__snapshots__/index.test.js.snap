// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MobileNumberWithCountryCode /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-jSFjdj FbyfL"
      data-testid="mobile-number-with-country-code"
    >
      <div
        class="ant-row ant-form-item sc-dlnjwi cPpdbK sc-eCApnc kzwLHv"
        data-testid="form-item"
      >
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                autocomplete="off"
                class="ant-input sc-gtsrHT nnNxg"
                data-testid="code-input"
                id="countryCode"
                maxlength="4"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-row ant-form-item sc-dlnjwi cPpdbK sc-gKAaRy kJAoRv"
        data-testid="form-item"
      >
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <span
                class="ant-input-affix-wrapper sc-gtsrHT nnNxg sc-fujyAs jFNWMi"
              >
                <input
                  autocomplete="off"
                  class="ant-input"
                  data-testid="mobile-input"
                  id="phone"
                  placeholder="contact_no"
                  type="number"
                  value=""
                />
                <span
                  class="ant-input-suffix"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
